src/sysupdate/sysupdate-feature.c

```diff
@@ -0,0 +1,118 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include "conf-parser.h"
+#include "hash-funcs.h"
+#include "path-util.h"
+#include "sysupdate-feature.h"
+#include "sysupdate.h"
+#include "web-util.h"
+
+static Feature *feature_free(Feature *f) {
+        free(f->id);
```

Comment:
please add `if (!f) return NULL;` at the top before dereffing `f`. We want that our destructors are generally happy with a NULL argument